docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,640 @@
+---
+
+# HttpWebRequest to HttpClient Migration Guide
+
+This document aims to guide developers through the process of migrating from <xref:System.Net.HttpWebRequest>, <xref:System.Net.ServicePoint>, and <xref:System.Net.ServicePointManager> to <xref:System.Net.Http.HttpClient>. The migration is necessary due to the obsolescence of the older APIs and the numerous benefits offered by <xref:System.Net.Http.HttpClient>, including improved performance, better resource management, and a more modern and flexible API design. By following the steps outlined in this document, developers will be able to transition their codebases smoothly and take full advantage of the features provided by <xref:System.Net.Http.HttpClient>.
+
+> [!WARNING]
+> Migrating from `HttpWebRequest`, `ServicePoint`, and `ServicePointManager` to `HttpClient` is not just a "nice to have" performance improvement. It's crucial to understand that the existing `WebRequest` logic's performance is likely to degrade significantly once you move to .NET Core. This is because `WebRequest` is maintained as a minimal compatibility layer, which means it lacks many optimizations, such as connection reuse in numerous cases. Therefore, transitioning to `HttpClient` is essential to ensure your application's performance and resource management are up to modern standards.
+
+## Migrating from <xref:System.Net.HttpWebRequest> to <xref:System.Net.Http.HttpClient>
+
+Let's start with some examples:
+
+**Simple GET Request Using <xref:System.Net.HttpWebRequest>**
+
+Here's an example of how the code might look:
+
+```c#
+using System.Net;
+
+HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
+using WebResponse response = await request.GetResponseAsync();
+```
+
+**Simple GET Request Using <xref:System.Net.Http.HttpClient>**
+
+Here's an example of how the code might look:
+
+```c#
+using System.Net.Http;
+
+HttpClient client = new();
+using HttpResponseMessage message = await client.GetAsync(uri);
+```
+
+**Simple POST Request Using <xref:System.Net.HttpWebRequest>**
+
+Here's an example of how the code might look:
+
+```c#
+using System.Net;
+
+HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
+request.Method = "POST";
+request.ContentType = "text/plain";
+await using Stream stream = await request.GetRequestStreamAsync();
+await stream.WriteAsync("Hello World!"u8.ToArray());
+using WebResponse response = await request.GetResponseAsync();
+Memory<byte> buffer = new byte[1024];
+await using Stream responseStream = response.GetResponseStream();
+```
+
+**Simple POST Request Using <xref:System.Net.Http.HttpClient>**
+
+Here's an example of how the code might look:
+
+```c#
+using System.Net.Http;
+
+HttpClient client = new();
+using HttpResponseMessage responseMessage = await client.PostAsync(uri, new StringContent("Hello World!"));
+```
+
+## HttpWebRequest to HttpClient, SocketsHttpHandler Migration Guide
+
+| <xref:System.Net.HttpWebRequest> Old API | New API | Notes |
```

Comment:
Nit: wording is inconsistent between "No direct equivalent API" and "No equivalent API"

Reply:
It's not inconsistent actually, `No direct equivalent API` means there is a workaround/different shaped usable API, `No equivalent API` means there is no workaround/usable API